src/apps/vpn/addons/addon.cpp

```diff
@@ -357,19 +358,19 @@ Addon* Addon::create(QObject* parent, const QString& manifestFileName) {
   Addon* addon = nullptr;
 
   if (type == "i18n") {
-    addon = new AddonI18n(parent, manifestFileName, id, name);
```

Comment:
I don't like this too much. Because you are checking something here and then you use it elsewhere extracting the same keys, without checking them.

Reply:
What about if you create the addon state after this block? See the condition block. This way you can use addon as the parent.

Reply:
Yes, I agree. I'll change it.